pubsub.go

```diff
@@ -356,7 +361,11 @@ func WithDiscovery(d discovery.Discovery, opts ...DiscoverOpt) Option {
 // WithEventTracer provides a tracer for the pubsub system
 func WithEventTracer(tracer EventTracer) Option {
 	return func(p *PubSub) error {
-		p.tracer = &pubsubTracer{tracer: tracer, pid: p.host.ID(), msgID: p.msgID}
+		if tracer != nil {
+			p.tracer.tracer = tracer
```

Comment:
I think `p.tracer` may be nil here - I get a nil-dereference when I pass a tracer into this branch, but it works if I pass nil

Reply:
oops, wrong check. it should be `if p.tracer != nil`....

fixing.

Reply:
fixed.